quickwit/quickwit-metastore/src/error.rs

```diff
@@ -48,6 +48,13 @@ pub enum MetastoreError {
     #[error("Failed to deserialize index metadata: `{message}`")]
     InvalidManifest { message: String },
 
+    /// Something has gone very wrong.
+    /// In theory this error should never be returned as doing so
+    /// indicates that the operation is attempting to perform an action
+    /// on the metastore state that would violates the integrity of the metastore.
+    #[error("Cannot complete operation as it is incorrect: `{message}`")]
+    CorrectnessError { message: String },
```

Comment:
I'm not really happy with the description or name for this error, but I couldn't think of anything more appropriate.

Reply:
Extending on this as well, we could potentially return the failed split ids as part of this error but the behaviours between the Postgres and file-backed metastore would be different unless we made the file-backed store eagerly try update the splits and then return the error.